scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,11 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    kwargs['flags'] = [encoding]
```

Comment:
What does this do?

Reply:
This adds an entry to the kwargs dict for 'flags' and places the encoding that the response was in as a list as its value.